File: project-001/bank-collection-service/dto/user_record_dto.go
package dto

type UserRecordDTO struct {
	ID              int64      `json:"id"`
	FirstName       string     `json:"firstName"`
	LastName        string     `json:"lastName"`
	Gender          string     `json:"gender"`
	IsTransferPin   bool       `json:"_transfer_pin"`
	Locked          bool       `json:"locked"`       // Omit if nil
	ReferralCode    string     `json:"referralCode,omitempty"`    // Omit if empty
	IsBlocked       bool       `json:"blocked"`
	BlockedDuration *int64     `json:"blockedDuration,omitempty"` // Omit if nil
	BlockedUntil    *string    `json:"blockedUntil,omitempty"`    // Omit if nil
	BlockedReason   *string    `json:"blockedReason,omitempty"`   // Omit if nil
	TotalRefs       *string    `json:"totalRefs,omitempty"`       // Omit if nil
	Notifications   *string    `json:"notifications,omitempty"`  // Omit if nil
	ReferralLink    string     `json:"referralLink,omitempty"`    // Omit if empty
	Photo           *string    `json:"photo,omitempty"`           // Omit if nil
}

File: fullstack-fintech-app/resiliences-service/DTOs/UserRecordDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace resiliences_service.DTOs
{
    public class UserRecordDTO
    {
        public long Id { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? Gender { get; set; }
        public string? Country { get; set; }
        public string? City { get; set; }
        public bool TransferPinSet { get; set; }
        public bool Locked { get; set; }
        public DateTime? LockedAt { get; set; }
        public string? ReferralCode { get; set; }
        public bool Blocked { get; set; }
        public long? BlockedDuration { get; set; }
        public string? BlockedUntil { get; set; }
        public string? BlockedReason { get; set; }
        public string? TotalReferers { get; set; }
        public string? ReferralLink { get; set; }
        public string? Photo { get; set; }
    }
}